test/integration/src/test/java/io/pravega/test/integration/BoundedStreamReaderTest.java

```diff
@@ -230,6 +250,128 @@ public void testBoundedStreamWithScaleTest() throws Exception {
+
+        //6. Continue reading on the reader where event data 1 is observed.
+        if (event4Reader1.equals(getEventData.apply(1))) {
+            assertEquals(getEventData.apply(1), reader1.readNextEvent(10000).getEvent());
+            EventRead<String> stringEventRead = reader1.readNextEvent(10000);
+            assertEquals(getEventData.apply(3), stringEventRead.getEvent());
+            reader1.close();
+            readerGroup.readerOffline("readerId1", stringEventRead.getPosition());
+
+        } else {
+            assertEquals(getEventData.apply(1), reader2.readNextEvent(10000).getEvent());
+            EventRead<String> stringEventRead = reader2.readNextEvent(10000);
+            assertEquals(getEventData.apply(3), stringEventRead.getEvent());
+            reader2.close();
+            readerGroup.readerOffline("readerId2", stringEventRead.getPosition());
+        }
+
+        StreamCut streamCut = readerGroup.getStreamCuts().get(Stream.of(SCOPE, STREAM1));
+
+        Map<Segment, Long> segmentMap = streamCut.asImpl().getPositions();
+        assertTrue(segmentMap.containsKey(new Segment(SCOPE, STREAM1, computeSegmentId(0, 0))));
+        assertTrue(segmentMap.containsKey(new Segment(SCOPE, STREAM1, computeSegmentId(3, 1))));
+        // Segment 0 -> offset 0;
+        // Segment 1 -> offset 30L
+        reader1.close();
+        reader2.close();
+        readerGroup.close();
+
+        ReaderGroupConfig readerGroupCfg2 = ReaderGroupConfig.builder().disableAutomaticCheckpoints()
+                                                             .stream(Stream.of(SCOPE, STREAM1), streamCut)
+                                                             .build();
+        groupManager.createReaderGroup("group2", readerGroupCfg2);
+        ReaderGroup readerGroup2 = groupManager.getReaderGroup("group2");
+
+        //Create two readers
+        @Cleanup
+        EventStreamReader<String> reader3 = clientFactory.createReader("readerId3", "group2", serializer,
+                                                                       ReaderConfig.builder().build());
+        @Cleanup
+        EventStreamReader<String> reader4 = clientFactory.createReader("readerId4", "group2", serializer,
+                                                                       ReaderConfig.builder().build());
+
+        String reader3Event = reader3.readNextEvent(15000).getEvent();
+        String reader4Event = reader4.readNextEvent(15000).getEvent();
+
+        log.info("=======================>");
+        if (reader3Event.equals(getEventData.apply(0))) {
+            // reader3 is the one reading from S0
+            String event = reader3.readNextEvent(15000).getEvent();
+            assertEquals(getEventData.apply(0), event);
+            // invoking read here should trigger readers from S2
+            event = reader3.readNextEvent(30000).getEvent();
+            assertEquals(getEventData.apply(2), event); // this fails.
```

Comment:
Observing that events from Segment 2 is not read...

Reply:
Are you observing a null or what?